layersvt/CMakeLists.txt

```diff
@@ -1,107 +1,33 @@
-file(GLOB IMAGES
-   "${PROJECT_SOURCE_DIR}/layersvt/images/*"
-)
+file(GLOB IMAGES "${PROJECT_SOURCE_DIR}/layersvt/images/*")
 file(COPY ${IMAGES} DESTINATION ${CMAKE_BINARY_DIR}/layersvt/images)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-    add_definitions(-DVK_USE_PLATFORM_WIN32_KHR -DVK_USE_PLATFORM_WIN32_KHX -DWIN32_LEAN_AND_MEAN -DNOMINMAX)
-    set(DisplayServer Win32)
-elseif(CMAKE_SYSTEM_NAME STREQUAL "Android")
+if(WIN32)
+    add_compile_definitions(VK_USE_PLATFORM_WIN32_KHR)
+    # Minimize what Windows.h leaks
+    add_compile_definitions(NOMINMAX WIN32_LEAN_AND_MEAN)
+elseif(ANDROID)
     add_compile_definitions(VK_USE_PLATFORM_ANDROID_KHR)
-elseif(CMAKE_SYSTEM_NAME MATCHES "Linux|BSD|DragonFly|GNU")
-    option(BUILD_WSI_XCB_SUPPORT "Build XCB WSI support" ON)
-    option(BUILD_WSI_XLIB_SUPPORT "Build Xlib WSI support" ON)
-
-    find_package(PkgConfig REQUIRED QUIET) # Use PkgConfig to find Linux system libraries
-
-    if (BUILD_WSI_XCB_SUPPORT)
-        pkg_check_modules(XCB REQUIRED QUIET IMPORTED_TARGET xcb)
-    endif()
-
-    if (BUILD_WSI_XLIB_SUPPORT)
-        pkg_check_modules(X11 REQUIRED QUIET IMPORTED_TARGET x11)
-    endif()
-
-    if (BUILD_WSI_XCB_SUPPORT)
-        add_definitions(-DVK_USE_PLATFORM_XCB_KHR -DVK_USE_PLATFORM_XCB_KHX)
-        set(DisplayServer Xcb)
-    endif()
-
-    if (BUILD_WSI_XLIB_SUPPORT)
-        if (NOT DisplayServer)
-            set(DisplayServer Xlib)
-        endif()
-        add_definitions(-DVK_USE_PLATFORM_XLIB_KHR -DVK_USE_PLATFORM_XLIB_KHX)
-    endif()
```

Comment:
This causes api dump to no longer dump xlib create surface. And while you are at it, enable wayland for api dump so it dumps that call as well. Wayland has the benefit of not including any header files (all types used are opaque pointers so are implicitly forward declared).

Reply:
API dump code for wayland is broken. I have re-enabled x11 though.